localstack/services/apigateway/apigateway_listener.py

```diff
@@ -742,22 +801,40 @@ def invoke_rest_api_integration_backend(
                 target = kinesis_listener.ACTION_LIST_STREAMS
             else:
                 LOG.info(
-                    "Unexpected API Gateway integration URI '%s' for integration type %s",
-                    uri,
-                    integration_type,
+                    f"Unexpected API Gateway integration URI '{uri}' for integration type {integration_type}",
                 )
                 target = ""
 
-            # apply request templates
-            new_data = apply_request_response_templates(
-                data, integration.get("requestTemplates"), content_type=APPLICATION_JSON
-            )
+            try:
+                json_str = json.dumps(data) if isinstance(data, (dict, list)) else to_str(data)
+                json_str = apply_template(
+                    integration,
+                    "request",
+                    json_str,
+                    path_params=path_params,
+                    query_params=query_string_params,
+                    headers=headers,
+                )
+                payload = json.loads(json_str)
+            except Exception as e:
+                LOG.error("Unable to convert API Gateway payload to str", e)
+                raise
+
             # forward records to target kinesis stream
-            headers = aws_stack.mock_aws_request_headers(service="kinesis")
+            headers = aws_stack.mock_aws_request_headers(
+                service="kinesis", region_name=invocation_context.region_name
```

Comment:
Out of curiosity: was this change (setting the `region_name`) required? If so, then it could point at a more general issue that the region is not properly configured in the thread-local request context for incoming API GW requests..?

Reply:
Yes, it is required, it was defaulting to `east-1` because the lookup failed - sometimes it's correctly configured, it certainly something to improve. However, is always correct in the invocation context.